group.go

```diff
@@ -21,202 +21,182 @@
 package pitaya
 
 import (
-	"sync"
-	"sync/atomic"
+	"context"
 
+	"github.com/coreos/etcd/clientv3"
+	"github.com/topfreegames/pitaya/config"
 	"github.com/topfreegames/pitaya/constants"
+	"github.com/topfreegames/pitaya/groups"
 	"github.com/topfreegames/pitaya/logger"
-	"github.com/topfreegames/pitaya/session"
-	"github.com/topfreegames/pitaya/util"
 )
 
-const (
-	groupStatusWorking = 0
-	groupStatusClosed  = 1
-)
+// Group represents an agglomeration of UIDs which is used to manage
+// users. Data sent to the group will be sent to all users in it.
+
+var groupServiceInstance groups.GroupService
 
-// SessionFilter represents a filter which is used to filter sessions when Multicast,
-// the session will receive the message when the filter returns true.
-type SessionFilter func(*session.Session) bool
-
-// Group represents a session group which is used to manage a number of
-// sessions, data sent to the group will be sent to all sessions in it.
-type Group struct {
-	mu       sync.RWMutex
-	status   int32                       // channel current status
-	name     string                      // channel name
-	sessions map[string]*session.Session // session id map to session instance
-}
-
-// NewGroup returns a new group instance
-func NewGroup(n string) *Group {
-	return &Group{
-		status:   groupStatusWorking,
-		name:     n,
-		sessions: make(map[string]*session.Session),
+// InitGroups should be called ONCE in the beginning of the application, if it is intended to use groups functionality
+func InitGroups(config *config.Config, clientOrNil *clientv3.Client) {
```

Comment:
Here you should receive `groupServiceOrNil` and create a EtcdGroupService if nil